streamlit_app/app.py

```
@@ -0,0 +1,259 @@
+
+def _get_run_state() -> tuple[str, bool]:
+    run_state = st.session_state.run_state
+    is_running = run_state == RUN_STATE_RUNNING
+    return run_state, is_running
+
+
+def _reject_upload(save_path: Path, missing: list[str]) -> None:
+    save_path.unlink(missing_ok=True)
+    aliases = {c: COLUMN_NAME_CONVERSION[c] for c in missing}
+    st.error(
+        f"Upload rejected — required field(s) not found: "
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)